### PR TITLE
Eng 4875  suggestion helper service

### PR DESCRIPTION
**Pull request by melisa-writer**
**Old:**
- README reupload
- avoid double masking reupload

**New:**
- rewriting fit.py (batch prediction, gpu support, multi token ranking, token guessing, pyfunctional syntax)
- tests.py updates
- delemmatize - set to optional

### CODE REVIEW

File: fitbert/fitb.py [hunk 1]
```diff
@@ -35,90 +40,196 @@ def softmax(x):
...
+        masked_tokens = tokens[:]
+        masked_tokens[mask_ind] = "[MASK]"
+        masked_tokens = ["[CLS]"] + masked_tokens + ["[SEP]"]
+        masked_ids = self.tokenizer.convert_tokens_to_ids(masked_tokens)
+        return masked_ids
+
+    def _get_sentence_probability(self, sent: str) -> float:
+
+        tokens = self.tokenizer.tokenize(sent)
+        input_ids = (
```
Comment (sam-qordoba) [inline fitbert/fitb.py]: I'm trying to learn `seq`... does wrapping it with parens make it a `tuple`? something else?

File: fitbert/fitb.py [hunk 2]
```diff
@@ -35,90 +40,196 @@ def softmax(x):
...
 
     def _delemmatize_options(self, options: List[str]) -> List[str]:
-        for word in options:
-            words_with_shared_lemma = self.delemmatizer(word)
-            for w in words_with_shared_lemma:
-                if w not in options:
-                    options.append(w)
+        options = (
+            seq(options[:])
+            .map(lambda x: self.delemmatizer(x))
```
Comment (sam-qordoba) [inline fitbert/fitb.py]: probably can replace `.map().flatten()` with [`.flat_map()`](https://docs.pyfunctional.org/en/master/functional.html#functional.pipeline.Sequence.flat_map)

**Review by sam-writer — APPROVED** [review]
nothing blocking, though I'd like to understand the thing where you wrap `seq(whatever).stuff()` in parens.

File: fitbert/fitb.py (hunk 1 above)
Comment (melisa-qordoba) [reply fitbert/fitb.py]: Hi, from pyfunctional docs: 

`seq = Stream()`

where: 

Stream is a class with `__call__` so we can use syntax `seq(...)`:

        Create a Sequence using a sequential ExecutionEngine.
        If args has more than one argument then the argument list becomes the sequence.
        If args[0] is primitive, a sequence wrapping it is created.
        If args[0] is a list, tuple, iterable, or Sequence it is wrapped as a Sequence.

File: fitbert/fitb.py (hunk 2 above)
Comment (melisa-qordoba) [reply fitbert/fitb.py]: Checked. Nice!!!

**Pull request merged**